pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -763,7 +763,7 @@ func (s *DiscoveryServer) pushEds(push *model.PushContext, con *XdsConnection, e
 		var l *xdsapi.ClusterLoadAssignment
 		var ok bool
 		// decide which to use based on presence of Sidecar.
-		if sidecarScope == nil || sidecarScope.Config == nil || len(pilot.DisableEDSIsolation) != 0 {
```

Comment:
This is the right fix..

Reply:
Bookinfo seems to work with and without this change, but tests are failing if we remove this. I have dug into it for a bit but have no clue why. Any ideas @rshriram? `AddInstance` doesn't seem to do anything when using clusterIsolated, although `SetEndpoints` does.

Reply:
I think the tests need a default sidecar scope.. Pilot code looks for a sidecar scope and if not found, it creates one in the push.initSidecarScope.. The one created by default wont have the config field set. I think the memregistry might be bypassign that code..

Reply:
As far as I can tell a default sidecar scope is created (in `push.initSidecarScope` from the test)

Reply:
The fix works without this change. From what I can tell, if we keep `|| sidecarScope.Config == nil`, we will default to the legacy version, and if we remove it we will use cluster isolation by default. 
So currently cluster isolation is not used if they dont have a Sidecar. Given this is a new feature, I don't think we want to remove this. Removing it also causes the tests to fail, and I have not be successful fixing them.

The bookinfo demo works with or without this check (because of the change below).

@rshriram you said this is the right fix. Do you think it is ok to remove this change, or is there something else we should do?

Reply:
I think we should either use the isolation version or the legacy version when generating EDS, but not mixing both of them though it doesn't cause any problems at this moment. But it seems dangerous if there is any inconsistency between the two implementation, and if that happens it would be very hard to debug and solve.

@ymesika you added those code and the isolation version in the first place, could you help to make the test pass? Thanks.

Reply:
cc @costinm Any opinion on this?

I think we need to get *some* fix for this in 1.1 - we have had at least 4 issues come in about this same bug.

Reply:
This effectively remove the old cache - and we don't have any testing on the impact on scale. 
Let's be safe - people who enable sidecar ( including the global ) will use the new code path, and people without Sidecar will keep the legacy behavior.  Once we have on-demand or caching for new code path or proof that caching doesn't help - we can remove the legacy code. 

The code looks correct - and in theory the new code path will work great. Question is if it scales - and for that we need testing.